bookkeeper-server/src/main/java/org/apache/bookkeeper/client/PendingAddOp.java

```diff
@@ -48,15 +51,18 @@
  *
  *
  */
-class PendingAddOp implements WriteCallback, TimerTask {
+class PendingAddOp extends SafeRunnable implements WriteCallback, TimerTask {
     private final static Logger LOG = LoggerFactory.getLogger(PendingAddOp.class);
 
+    ByteBuf payload;
     ByteBuf toSend;
     AddCallback cb;
     Object ctx;
     long entryId;
     int entryLength;
-    int[] writeSet;
+
+    int writeQuorumSize;
+    int[] writeSet = new int[0];
```

Comment:
do we need this allocation of "new int[0]"?

Reply:
it avoids the need for a null check further down, and it only happens when the op is first created, so there's negligible cost for it.

Reply:
can you define a constant for `new int[0]`?